exercises/strings/countLetter/countLetter.js

```diff
@@ -22,14 +22,20 @@
  */
 
 function countLetter(string, letter) {
-  // This is your job. :)
+  
+  for(l of string) {
+    if(l === letter){
+      
+    }
+  }
+  return l.length
```

Comment:
Right now this is saying:

```text
look at each in the STRING, call each one L
  if L is LETTER then {
    do nothing
  }

return the length of L
```

So `l` is each individual letter of `string` in turn. `l.length` will always be `1` since `l` is a single letter. You look at each letter in `string`, one at a time, and check if a given letter `l` matches the user-supplied letter in `letter`. But when there's a match you don't do anything.

Consider the following:

```js
function countLetter(string, letter) {
  let letterCount = 0;

  for (let char of string) {
    if (char === letter) {
      letterCount = letterCount + 1; // increment the value of letterCount
    }
  }

  return letterCount
}
```

Here, we initialize a variable called `letterCount` to `0`, which is going to keep track of how many times we see `letter` in `string`. Every time we see it, we increment `letterCount` by `1`.

Using `char` vs `l` doesn't matter, it's just a placeholder name for each individual character in the `string`, accessed one at a time. We have to give it some kind of name so we can reference it and `char` (short for _character_) is less likely to be involved in some kind of typo than `l` (which often looks like `1` or `I`, depending on what font people use).

Reply:
There are other ways of incrementing a variable but they're all short-hand for `x = x + 1`. Here are the three ways:

```js
x = x + 1
x += 1
x++
```